libc/src/__support/big_int.h

```diff
@@ -285,7 +285,10 @@ LIBC_INLINE constexpr cpp::array<word, N> shift(cpp::array<word, N> array,
       return 0;
     if (i >= int(N))
       return is_neg ? cpp::numeric_limits<word>::max() : 0;
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wsign-conversion"
     return array[i];
```

Comment:
This is the only code that get flagged (atleast locally for an overlay build).

it was changed to use `size_t` in: https://github.com/llvm/llvm-project/pull/127523

but had to be reverted in: https://github.com/llvm/llvm-project/pull/129771#issue-2895388911


Discussion on why it needs `signed int` has begun here: https://github.com/llvm/llvm-project/pull/129771#issuecomment-2699452560

Reply:
Note: pragma(s) are only WIP for now, will be fixed in final version of PR based on above discussion

Reply:
This didn't work, as the error is at:
```
llvm-project/libc/src/__support/big_int.h:288:36: error: implicit conversion changes signedness: 'const int' to 'size_t' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    return static_cast<word>(array[i]);
```

But:

```suggestion
    return array[static_cast<size_t>(i)];
```

did work